Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -324,21 +324,44 @@ class PurchasesOrchestrator {
     }
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    func purchase(sk2Product product: SK2Product,
-                  discount: StoreProductDiscountType?) async throws -> PurchaseResultData {
-        let result: Result<(CustomerInfo, Bool), Error> = await self.purchase(sk2Product: product, discount: discount)
+    func purchase(
```

Comment:
This is basically the method below without the `do {} catch {}` and unnecessary conversion from/to `Result<>`